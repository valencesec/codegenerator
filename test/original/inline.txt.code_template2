// inline template: data file: "test/workdir/inline.yaml"
capitalCamelCase:
{{- range $listItem := $.listOfItems }}
List Item: {{ $listItem }}
List Item quoted: {{ $listItem | quote }}
List Item capitalCamelCase: {{ $listItem | capitalCamelCase }}
{{- end }}
orVoid:
{{ $.block.doesNotExist }}
{{ $.block.doesNotExist | orVoid }}
{{ $.block.aKey | orVoid }}
goType:
number: {{ "number" | goType }}
string: {{ "string" | goType }}
boolean: {{ "boolean" | goType }}
unknown: {{ "unknown" | goType }}
SomethingElse: {{ "SomethingElse" | goType }}
goTypeWithModule:
number: {{ "number" | goTypeWithModule "*model." }}
string: {{ "string" | goTypeWithModule "*model." }}
boolean: {{ "boolean" | goTypeWithModule "*model." }}
SomethingElse: {{ "SomethingElse" | goTypeWithModule "*model." }}
typescriptTypeWithModule:
number: {{ "number" | typescriptTypeWithModule "model." }}
string: {{ "string" | typescriptTypeWithModule "model." }}
boolean: {{ "boolean" | typescriptTypeWithModule "model." }}
unknown: {{ "unknown" | typescriptTypeWithModule "model." }}
any: {{ "any" | typescriptTypeWithModule "model." }}
SomethingElse: {{ "SomethingElse" | typescriptTypeWithModule "model." }}
isBasicType:
number: {{ if isBasicType "number" }}YES{{ else }}NO{{ end }}
string: {{ if isBasicType "string" }}YES{{ else }}NO{{ end }}
boolean: {{ if isBasicType "boolean" }}YES{{ else }}NO{{ end }}
SomethingElse: {{ if isBasicType "SomethingElse" }}YES{{ else }}NO{{ end }}
uppercaseToCapitalized:
PUT: {{ "PUT" | uppercaseToCapitalized }}
replace:
a/b/${id}: {{ "a/b/${id}" | replace "${id}" ":id" }}
split:
a/b/c: {{ index (split "/" "a/b/c") 1 }}
splitIndexOf:
a/b/c: {{ splitIndexOf "/" "b" "a/b/c" }}
splitIndexOfNegative:
a/b/c: {{ splitIndexOfNegative "/" "b" "a/b/c" }}
lowerNoUnderscore:
ABC_DEF_123: {{ "ABC_DEF_123" | lowerNoUnderscore }}
camelCaseNoUnderscore:
ABC_DEF_123: {{ "ABC_DEF_123" | camelCaseNoUnderscore }}
capitalCamelCaseNoUnderscore:
ABC_DEF_123: {{ "ABC_DEF_123" | capitalCamelCaseNoUnderscore }}
upperSpaceToUnderscore:
a b 345 c: {{ "a b 345 c" | upperSpaceToUnderscore}}
rustType:
number: {{ "number" | rustType }}
string: {{ "string" | rustType }}
boolean: {{ "boolean" | rustType }}
SomethingElse: {{ "SomethingElse" | rustType }}
camelCaseToLowerSnakeCase:
firstSecondThird: {{ "firstSecondThird" | camelCaseToLowerSnakeCase }}
trimPrefix: {{ "prefixThing" | trimPrefix "prefix" }}
trimSuffix: {{ "ThingSuffix" | trimSuffix "Suffix" }}
cutTakeBefore: {{ "/v2/abc/${id}/efg" | cutTakeBefore "${id}" }}
cutTakeAfter: {{ "/v2/abc/${id}/efg" | cutTakeAfter "${id}" }}
stringSliceContains true: {{ if stringSliceContains $.listOfItems "firstItem" }}affirmative{{else}}nope{{end}}
stringSliceContains false: {{ if stringSliceContains $.listOfItems "doesntExist" }}nope{{else}}affirmative{{end}}
stringSliceOnlyContainsEntriesFromStringSlice: {{ if stringSliceOnlyContainsEntriesFromStringSlice $.superSetOfString $.subsetOfString }}affirmative{{else}}nope{{end}}
stringSliceOnlyContainsEntriesFromStringSlice: {{ if stringSliceOnlyContainsEntriesFromStringSlice $.superSetOfString $.partialIntersect }}nope{{else}}affirmative{{end}}
// inline template end
