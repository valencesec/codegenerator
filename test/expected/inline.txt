capitalCamelCase:
List Item: firstItem
List Item quoted: "firstItem"
List Item capitalCamelCase: FirstItem
List Item: secondItem
List Item quoted: "secondItem"
List Item capitalCamelCase: SecondItem
orVoid:
<no value>
void
aValue
goType:
number: int64
string: string
boolean: bool
unknown: interface{}
SomethingElse: SomethingElse
goTypeWithModule:
number: int64
string: string
boolean: bool
SomethingElse: *model.SomethingElse
typescriptTypeWithModule:
number: number
string: string
boolean: boolean
unknown: unknown
any: any
SomethingElse: model.SomethingElse
isBasicType:
number: YES
string: YES
boolean: YES
SomethingElse: NO
uppercaseToCapitalized:
PUT: Put
replace:
a/b/${id}: a/b/:id
split:
a/b/c: b
splitIndexOf:
a/b/c: 1
splitIndexOfNegative:
a/b/c: -2
lowerNoUnderscore:
ABC_DEF_123: abcdef123
camelCaseNoUnderscore:
ABC_DEF_123: abcDef123
capitalCamelCaseNoUnderscore:
ABC_DEF_123: AbcDef123
upperSpaceToUnderscore:
a b 345 c: A_B_345_C
rustType:
number: i64
string: String
boolean: bool
SomethingElse: SomethingElse
camelCaseToLowerSnakeCase:
firstSecondThird: first_second_third
trimPrefix: Thing
trimSuffix: Thing
cutTakeBefore: /v2/abc/
cutTakeAfter: /efg
stringSliceContains true: affirmative
stringSliceContains false: affirmative
